### DISCUSSION TITLE
Using dev-only dependencies in next.config.js

### DISCUSSION THREAD

**Discussion by Juddling** [vercel/next.js#14615] [Help]
I'm looking for a way to set up custom webpack configuration in my `next.config.js` without needing to list it as a production dependency in my `package.json` file.

I believe the pattern recommended in the [readme for @next/bundle-analyzer](https://github.com/vercel/next.js/tree/canary/packages/next-bundle-analyzer) for installing the bundle analyzer could be changed so that the package is only required as a dev dependency, and wouldn't fail in prod at runtime, if installed with `yarn install --production`

Would love to hear if anyone has found a pattern which supports loading dev dependencies only at build time!

**Comment by ethanryan** [comment]
Hi @Juddling, I was curious about this too, and tried including @next/bundle-analyzer as a dev dependency instead of a dependency to see if it would reduce our app's bundle size and our first load JS shared by all, but unfortunately it did not.

**Comment by ijjk** [comment]
Hi, you can place the requires/config for dev only dependencies inside of a `if (process.env.NODE_ENV === 'development') {}` check which should prevent them from being loaded in production.
